Adyen/Model/Amount.swift

```diff
@@ -119,3 +115,54 @@ extension Amount: Comparable {
     }
 
 }
+
+/// :nodoc:
+public struct AmountComponents {
+    
+    /// :nodoc:
+    public init(amount: Amount) {
+        if let comps = Self.extractAmountComponents(from: amount.formatted) {
+            (self.formattedCurrencySymbol, self.formattedValue) = comps
+        } else {
+            self.formattedValue = String(amount.value)
+            self.formattedCurrencySymbol = amount.currencyCode
+        }
+    }
+    
+    /// :nodoc:
+    public let formattedValue: String
+    
+    /// :nodoc:
+    public let formattedCurrencySymbol: String
+    
+    private static func extractAmountComponents(
+        from formattedString: String
+    ) -> (currency: String, value: String)? {
+        guard let regexp = try? NSRegularExpression(
+            pattern: "(\\d+(?:[.,\\s]\\d+)+)",
+            options: []
+        ), let match = regexp.firstMatch(
+            in: formattedString,
+            options: [],
+            range: NSRange(location: 0, length: formattedString.count)
+        ) else {
+            return nil
+        }
```

Comment:
using `NSRegularExpression` and `regexp.firstMatch ` is very verbose, you can use this:

```
formattedString.range(of: regularExpression, options: .regularExpression
```

this is much less code, check how `RegularExpressionValidator` does it.

Reply:
great idea 👍